docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+        }
+    }
+}
+```
+
+The containers you launch into this network must reside on the same Docker host.  Each container in the network can immediately communicate with other containers in the network. Though, the network itself isolates the containers from external networks.
+
+![](images/bridge_network.png)
+
+Within of a user-defined bridge network, linking is not supported. You can expose and publish container ports on containers in this network. This is useful if you want make a portion of the `bridge` network available to an outside network.
+
+![](images/network_access.png)
+
+A bridge network is useful in cases where you want to run a relatively small network on a single host. You can, however, create significantly larger networks by creating an `overlay` network.
+
+
+### An overlay network
+
+Docker's `overlay` network driver supports multi-host networking natively
+out-of-the-box. This support is accomplished with the help of `libnetwork`, a
+built-in VXLAN-based overlay network driver, and Docker's `libkv` library.
+
+The `overlay` network requires a valid key-value store service. Currently, Docker's supports Consul, Etcd, Zookeeper (Distributed store) and
+BoltDB (Local store). Before creating a network you must install and configure your chosen key-value store service.  The Docker hosts that you intend to network and the service must be able to communicate.
+
+![](images/key_value.png)
+
+Each host in the network must run a Docker Engine instance. The easiest way to provision the hosts are with Docker Machine.
+
+![](images/engine_on_net.png)
+
+Using Docker Swarm you can quickly create a cluster which includes a discovery service as well.
+
+To create an overlay network, you configure options on  the `daemon` on each Docker Engine for use with `overlay` network. There are two options to set:
+
+| Option                           | Description                                               |
+|----------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.                 |
+| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+
+Create an `overlay` network on one of the machines in the Swarm.
+
+        $ docker network create -d overlay my-multi-host-network
+
+This results in a single network spanning multiple hosts. The `overlay` network provides complete isolation for the containers.
+
+![](images/overlay_network.png)
+
+Then, on each host, launch containers making sure to specify the network name.
+
+        $ docker run run -itd --net=mmy-multi-host-network busybox
+
+Once launched, each container has access to all the containers in the network regardless of which Docker host the container was launched on.
+
+![](images/overlay-network-final.png)
+
+If you would like to try this for yourself, see the [Getting started for overlay](get-started-overlay.md).
```

Comment:
To prevent confusion, we may want to use "overlay *networking*" to prevent mix ups with the overlay *graph driver* (this has lead to confusing conversations between maintainers sometimes as well)